### PR TITLE
Venue handling refactor

### PR DESCRIPTION
**Pull request by ammapspeople**
**What**

- Rewrote venue handling in the Map Template.

**Why**

- Several bugs related to venues made us do a more thorough refactor.

**How**

- A new custom hook, useCurrentVenue that isolates side effects to changing venue
- Refactor the naming of the timing controls in the useMapBoundsDeterminer hook to be more clear about what they are.
- Removed the venueState atom since it is not necessary, and having several separate states for the venue is confusing.
- Cleaned up unnecessary calls to set venue.
- Introduced a venueWasSelected boolean atom to indicate if the user has selected a venue. This is used to be able to switch the venue on the map.
- Only set the "AUSTINOFFICE" as venue if there is no apiKey and no venue set.

### CODE REVIEW

File: packages/map-template/src/hooks/useMapBoundsDeterminer.js [hunk 1]
```diff
@@ -74,23 +69,18 @@ const useMapBoundsDeterminer = () => {
...
     useEffect(() =>  {
         determineMapBounds();
-    }, [mapsIndoorsInstance, venue, venues, locationId, kioskOriginLocationId, pitch, bearing, startZoomLevel, categories]);
+    }, [mapsIndoorsInstance, currentVenueName, locationId, kioskOriginLocationId, pitch, bearing, startZoomLevel, categories]);
 
     /**
      * Based on the combination of the states for venueName, locationId & kioskOriginLocationId,
      * determine where to make the map go to.
      *
      * @param {string} [forcedVenue] - If set, this venue will be used instead of the current venue.
```
Comment (andreeaceachir142) [inline packages/map-template/src/hooks/useMapBoundsDeterminer.js:78]: I assume this param should be removed?

File: packages/map-template/src/components/Map/Map.jsx [hunk 2]
```diff
@@ -125,21 +123,21 @@ function Map({ onLocationClick, onVenueChangedOnMap, useMapProviderModule, onMap
      * When map position is known, run callback.
      */
     useEffect(() => {
```
[the remark this answers is not in the document]
Comment (ammapspeople) [reply packages/map-template/src/components/Map/Map.jsx:125]: Any suggestions of to what? Only the naming changed 🤔

[the remark this answers is not in the document]
Comment (ammapspeople) [reply packages/map-template/src/components/Map/Map.jsx:125]: I'll see to it.

File: packages/map-template/src/components/Map/Map.jsx [hunk 3]
```diff
@@ -125,21 +123,21 @@ function Map({ onLocationClick, onVenueChangedOnMap, useMapProviderModule, onMap
...
+            onMapPositionInvestigating();
         }
-    }, [mapPositionKnown]);
+    }, [mapPositionInvestigating]);
 
     /*
      * When venue is changed on the map, run callback.
      * Set the current venue name whenever the venue is changed on the map.
      */
     useEffect(() => {
```
[the remark this answers is not in the document]
Comment (ammapspeople) [reply packages/map-template/src/components/Map/Map.jsx:135]: 🫡

File: packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx [hunk 4]
```diff
@@ -88,9 +87,18 @@ function MapsIndoorsMap(props) {
         const showExternalIDsQueryParameter = getBooleanQueryParameter(queryStringParams.get('showExternalIDs'));
         const showRoadNamesQueryParameterBoolean = getBooleanQueryParameter(queryStringParams.get('showRoadNames'));
 
+        // Set the initial props on the Map Template component.
+
+        // For the apiKey and venue, set the venue to "AUSTINOFFICE" if the apiKey is "mapspeople3d" and no venue is provided. We want this as the default venue for the "mapspeople3d" apiKey.
+        const apiKey = props.supportsUrlParameters && apiKeyQueryParameter ? apiKeyQueryParameter : (props.apiKey || defaultProps.apiKey);
+        let venue = props.supportsUrlParameters && venueQueryParameter ? venueQueryParameter : (props.venue || defaultProps.venue);
+        if (apiKey === 'mapspeople3d' && !venue) {
```
[the remark this answers is not in the document]
Comment (ammapspeople) [reply packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx:95]: Both can be justified: 

1. We know we never need to change `apiKey`. 
2. We want to be able to change the value of `venue`.

File: packages/map-template/src/hooks/useMapBoundsDeterminer.js (hunk 1 above)
Comment (ammapspeople) [reply packages/map-template/src/hooks/useMapBoundsDeterminer.js:78]: 👀 🫡

File: packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx:95]: Alright, let's keep it then.

**Review by andreeaceachir142 — APPROVED** [review]
✅ 🆗 🆗 🆗

**Pull request merged**